gdx/src/com/badlogic/gdx/graphics/Cubemap.java

```diff
@@ -43,32 +44,49 @@
 	/** Enum to identify each side of a Cubemap */
 	public enum CubemapSide {
 		/** The positive X and first side of the cubemap */
-		PositiveX(0, GL20.GL_TEXTURE_CUBE_MAP_POSITIVE_X),
+		PositiveX(0, GL20.GL_TEXTURE_CUBE_MAP_POSITIVE_X, 0, -1, 0, 1, 0, 0),
 		/** The negative X and second side of the cubemap */
-		NegativeX(1, GL20.GL_TEXTURE_CUBE_MAP_NEGATIVE_X),
+		NegativeX(1, GL20.GL_TEXTURE_CUBE_MAP_NEGATIVE_X, 0, -1, 0, -1, 0, 0),
 		/** The positive Y and third side of the cubemap */
-		PositiveY(2, GL20.GL_TEXTURE_CUBE_MAP_POSITIVE_Y),
+		PositiveY(2, GL20.GL_TEXTURE_CUBE_MAP_POSITIVE_Y, 0, 0, 1, 0, 1, 0),
 		/** The negative Y and fourth side of the cubemap */
-		NegativeY(3, GL20.GL_TEXTURE_CUBE_MAP_NEGATIVE_Y),
+		NegativeY(3, GL20.GL_TEXTURE_CUBE_MAP_NEGATIVE_Y, 0, 0, -1, 0, -1, 0),
 		/** The positive Z and fifth side of the cubemap */
-		PositiveZ(4, GL20.GL_TEXTURE_CUBE_MAP_POSITIVE_Z),
+		PositiveZ(4, GL20.GL_TEXTURE_CUBE_MAP_POSITIVE_Z, 0, -1, 0, 0, 0, 1),
 		/** The negative Z and sixth side of the cubemap */
-		NegativeZ(5, GL20.GL_TEXTURE_CUBE_MAP_NEGATIVE_Z);
+		NegativeZ(5, GL20.GL_TEXTURE_CUBE_MAP_NEGATIVE_Z, 0, -1, 0, 0, 0, -1);
 
 		/** The zero based index of the side in the cubemap */
 		public final int index;
 		/** The OpenGL target (used for glTexImage2D) of the side. */
 		public final int glEnum;
+		/** The up vector to target the side. */
+		public final Vector3 up;
+		/** The direction vector to target the side. */
+		public final Vector3 direction;
 
-		CubemapSide (int index, int glEnum) {
+		CubemapSide (int index, int glEnum, float upX, float upY, float upZ,
+			float directionX, float directionY, float directionZ) {
 			this.index = index;
 			this.glEnum = glEnum;
+			this.up = new Vector3(upX, upY, upZ);
+			this.direction = new Vector3(directionX, directionY, directionZ);
 		}
 
 		/** @return The OpenGL target (used for glTexImage2D) of the side. */
 		public int getGLEnum () {
 			return glEnum;
 		}
+
+		/** @return The up vector of the side. */
+		public Vector3 getUp () {

```

Comment:
Could you change this and the direction getter to:

    public Vector3 getUp (Vector3 out) { return out.set(up); }